textures/metro_control_panel_001_pegboard.py

```diff
@@ -32,3 +34,29 @@ def _swap_pegs(pegs: Pegs, a: str, b: str) -> Pegs:
     return tuple(
         tuple(s.replace(a, placeholder).replace(b, a).replace(placeholder, b) for s in panel) for panel in pegs
     )
+
+
+def _markers_to_bool_list(peg_str: str) -> Iterable[bool]:
+    return map(lambda b: True if b == "M" else False, peg_str)
+
+
+def _pegs_to_bool_list(peg_str: str) -> Iterable[bool]:
+    return map(lambda b: False if b == "E" else True, peg_str)
+
+
+def _invert_bool_list(bits: Iterable[bool]) -> Iterable[bool]:
+    return map(lambda b: not b, bits)
+
+
+def _bools_to_nor_mask(peg_str: str) -> Iterable[Callable]:
+    def nor(a: bool, b: bool) -> bool:
+        return not (a or b)
+
+    def pass_thru(a: bool, b: bool) -> bool:
+        return a
+
+    return [nor if bit else pass_thru for bit in _markers_to_bool_list(peg_str)]
+
+
+def _bools_to_binary_str(bits: Iterable[bool]) -> str:
+    return insert_spaces("".join(str(int(b)) for b in bits), 8)
```

Comment:
```suggestion
    return insert_spaces("".join("1" if b else "0") for b in bits), 8)
```

Also, maybe make this generally available, add some docs and doctests.